attempt to fix DaoPatient stall during reCache()

sheridancbio opened the pull request:
- all primitive patient records extracted from record set before cancer studies are retrieved
- primitive patient records are processed after recordset/connection is closed (cancer study objects are retrieved and Patient model objects are created)
- also removed apparently unnecessary static initializer block in DaoGeneticProfile
- removed "throws DaoException" specifier from DaoCancerStudy.getCancerStudyByInternalId() (exception is no longer possible)
- unnecessary try/catch blocks removed in related classes
- cleaned up cachePatient() function to use only the passed Patient object, which contains the needed studyId in the CancerStudy member

These fixes, and this new branch is to attempt to resolve an intermittent stalling of the import process during the recaching step when using the jdbc datasource. This branch (importer-daopatient-stall-fix) will be kept current with master (rebasing as necessary) and will be used to build the import pipeline. When it appears that the intermittent stalling has been resolved, the code will be tested for use in the production web service portals and a PR for a merge into a release branch will be opened.

Changes proposed in this pull request:

averyniceday commented on core/src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java:
```diff
@@ -60,20 +83,20 @@ private static void clearCache()
             rs = pstmt.executeQuery();
-            ArrayList<Patient> list = new ArrayList<Patient>();
             while (rs.next()) {
-                Patient p = extractPatient(rs);
```
call to getCancerStudy separated out from result set - connection closed first

sheridancbio commented:
I have tested import using this codebase. I still saw 2 reCache reports:

angelicaochoa commented on core/src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java:
```diff
@@ -60,20 +83,20 @@ private static void clearCache()
     {
         clearCache();
-
+        Vector<PrimitivePatientRecord> patientRecordsRetrieved = new Vector<>();
```
just curious.. why a vector?

sheridancbio replied:
The Vector class is thread safe, and we use synchronized methods in DaoCancerStudy .. presumably to prevent two parallel thread from both calling reCacheAll.
So if multithreading is a worry, it seemed safer to stay with the threadsafe class (rather than ArrayList).

angelicaochoa replied:
@sheridancbio maybe you can add a comment saying so? that's useful to know. either in the pr or here at this line?

sheridancbio replied:
I will add comment to the code

sheridancbio replied:
Actually, on deeper reflection, this variable is already inside a synchronized function call. And as a local variable, it would only have visibility to a single thread anyway. To say that Vector is "thread safe" really means that the read/write methods are synchronized on the object ... but we don't need that inside a synchronized method really. So I'm going to switch it to ArrayList

sheridancbio commented:
I have changed the Vector into an ArrayList inside DaoPatient.reCache()
I have also added a second commit where a 4 second delay is added after closing the connection on an addCancerStudy() call before we reCacheAll() ... under the theory that perhaps a large transaction needs some time to be committed/cleared before we can try re-reading/re-caching studies.

sheridancbio commented:
4 second delay increased to 5 seconds (after checking with @averyniceday)
